sample/redo/lib/main.dart

```diff
@@ -128,24 +128,29 @@ class _MyHomePageState extends State<MyHomePage> {
   }
 
   Future<void> _load() async {
-    List<int> listIds = List.from((await _replicache.scan(prefix: '/list/'))
-        .map((item) => int.parse(item.id.substring('/list/'.length))));
+    await _replicache.query((tx) async {
+      final listIdScanItems = await tx.scan(prefix: '/list/');
+
+      List<int> listIds = List.from(listIdScanItems
+          .map((item) => int.parse(item.id.substring('/list/'.length))));
+
+      if ((_selectedListId == null || !listIds.contains(_selectedListId)) &&
```

Comment:
The rest of this should be outside the `query` callback.

Reply:
This was intentional. I wanted the two scan functions to be using the same tx. I will restructure it slightly.

Reply:
I might be confused...

I only see one scan().
 
Also query() impl should not really have side-effects (eg setState). It should return results and callee should used them.